<launch>

    <arg name="vertical_filter_pct" default="15.0" />

    <node pkg="icuas23_competition" type="image_narrower.py" name="image_narrower">
        <param name="vertical_filter_pct" type="double" value="$(arg vertical_filter_pct)"/>

        <remap from="in/image_raw" to="/red/camera/depth/image_raw"/>
        <remap from="in/camera_info" to="/red/camera/depth/camera_info"/>
        <remap from="out/image_raw" to="/red/camera_narrow/depth/image_raw"/>
        <remap from="out/camera_info" to="/red/camera_narrow/depth/camera_info"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

    <node pkg="nodelet" type="nodelet" name="nodelet1"
            args="load depth_image_proc/point_cloud_xyz nodelet_manager">
        <remap from="camera_info" to="/red/camera/depth/camera_info"/>
        <remap from="image_rect" to="/red/camera_narrow/depth/image_raw"/>
        <remap from="points" to="/red/camera_narrow/depth/points"/>
    </node>
</launch>
